src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/LinksMapper.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.companieshouse.filinghistory.consumer.delta;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.filinghistory.FilingHistoryItemDataLinks;
+
+@Component
+public class LinksMapper {
+
+    public FilingHistoryItemDataLinks map(final String companyNumber, final String transactionId) {
+        if (StringUtils.isBlank(companyNumber) || StringUtils.isBlank(transactionId)) {
+            throw new IllegalArgumentException("Cannot map self link with null or empty data");
```

Comment:
Log an error message here with a similar message. Can also add companyNumber to the DataLogMap here now we know its not blank.